lib/connection.js

```diff
@@ -227,7 +227,7 @@ C.open = function(allFields, openCallback0) {
       tunedOptions.channelMax =
         negotiate(fields.channelMax, allFields.channelMax);
       tunedOptions.heartbeat =
```

Comment:
I know this PR is old and @squaremo has reviewed / suggested a change, but I am thinking this is a simpler, clearer solution.
```suggestion
if (allFields.heartbeat !== 0) {
  tunedOptions.heartbeat =
    negotiate(fields.heartbeat, allFields.heartbeat);
}
```
This would also facilitate revert of the change @ line 192 above.